### PR TITLE
Upgrade algolia instantsearch

### LINKED ISSUE #1382: Use new Algolia Next.js integration to avoid full page reload on back button
Due to InstantSearch and Next.js routing not playing nicely together, we have some weird work around in place to prevent bugs. More specifically we're triggering a full page reload when the user clicks the browser back button, and this results in a suboptimal experience.

The new package released by Algolia  (https://www.npmjs.com/package/react-instantsearch-hooks-router-nextjs) seems to [address this issue](https://github.com/algolia/instantsearch/issues/5241#issuecomment-1431005385)

## Todo

Use the package and remove the full page reload behavior

### LINKED ISSUE #1487: Collections: Item-type name persists in page title after its removal
After removing the `Accessories` selection in the ui, the page-title still has `Accessories` appended. The URL, breadcrumbs, and filtered items are reset. This is happening when the item-type page is directly entered or reloaded. 
image

Working:
1. Go to https://www.ifixit.com/Parts/Phone
2. Add an item type selection in the UI
3. Remove the item type selection
4. Title is updated

Not working:
1. Go to https://www.ifixit.com/Parts/Phone
2. Add an item type selection in the UI
3. Refresh the page (or just enter the item-type page directly)
5. Remove the item type selection
6. Title is not updated

Not sure if users are refreshing the page too much, but we link directly to item type pages from our iFixit device pages and Google. 

<details><summary>Screen-recording of the above steps</summary>

https://user-images.githubusercontent.com/72166715/226480224-fa063ebb-ef42-4039-adc2-59ecb4b2c3c4.mp4

</details>

### PR DESCRIPTION
**Pull request by federicobadini**
closes #1382 
closes #1487 

This PR introduces the new `react-instantsearch-hooks-router-nextjs` package.
With this new package, the nextjs router now plays nicely with Algolia.

On the Algolia side there is still [this pending issue](https://github.com/algolia/instantsearch/issues/5571) that has been mitigated by a little hack

### QA

1. Visit Vercel preview
2. Verify that the browser back button is working correctly
3. Verify that the item type name persisting in the hero section issue is now fixed

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

The back button is still working as expected ✅ 
The title is now updated and https://github.com/iFixit/react-commerce/issues/1487 no longer persists ✅ 

### Testing Procedure:

1. Go to the [Vercel Preview](https://react-commerce-git-fix-item-type-name-persisting-863b89-ifixit.vercel.app/Parts/iPhone)
2. Add an item type selection in the UI _(e.g. Accessories)_
3. Refresh the page
4. Remove the item type selection

> Ensure that the title of the page is updated correctly

File: frontend/components/common/InstantSearchProvider.tsx [hunk 1]
```diff
@@ -57,196 +56,207 @@ export function InstantSearchProvider({
...
+            ).trim();
+         }
 
-            const filterObject =
-               typeof filter === 'object' && !Array.isArray(filter)
-                  ? filter
-                  : {};
+         return {
+            q: String(q || ''),
+            p: typeof p === 'string' ? parseInt(p) : undefined,
```
Comment (masonmcelvain) [inline frontend/components/common/InstantSearchProvider.tsx:182]: Dropping the non-negative check on `p` can cause the page to 500, for example visiting `/Parts?p=-1`

**Review by masonmcelvain — CHANGES_REQUESTED** [review]
CR ✌🏻 I'm excited to see this change, thank you! dev_block ✌🏻 on the negative page param bug. Up to you if you'd like to remove `deviceItemType` from the productList model or not, we could always do it in another PR.

**Comment by federicobadini** [comment]
un_dev_block ✌🏻
Cleaned deviceItemType and restored the non-negative check

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Thanks for the changes!

**Pull request merged**